File: input/test_02.rb
require 'rails_helper'

RSpec.describe 'API', type: :request do
  it 'returns details of an item' do
    get '/api/items/1'
    expect(response).to have_http_status(200)
    expect(response.body).to include('item1')
  end

  it 'updates an item' do
    put '/api/items/1', params: { item: { name: 'updated_item1' } }
    expect(response).to have_http_status(200)
    expect(response.body).to include('updated_item1')
  end
end

File: output/test_02.py
import requests
import requests_mock

def test_api():
    with requests_mock.Mocker() as mocker:
        mocker.get('http://localhost:3000/api/items/1', status_code=200, json=["item1"])
        mocker.put('http://localhost:3000/api/items/1', status_code=200, json=["updated_item1"])
        response = requests.get('http://localhost:3000/api/items/1')
        assert response.status_code == 200
        assert all(item in response.json() for item in ["item1"])

        response = requests.put('http://localhost:3000/api/items/1', json={"item": {"name": "updated_item1"}})
        assert response.status_code == 200
        assert all(item in response.json() for item in ["updated_item1"])

